addons/arsenal/functions/fnc_attributeImport.sqf

```diff
@@ -15,32 +16,43 @@
  * Public: No
  */
 
-params ["_controlsGroup"];
-
 private _importList = call compile copyFromClipboard;
 
 // Verify import list is in correct format
 if (isNil "_importList" || {!(_importList isEqualType [])} || {!(_importList isEqualTypeAll "")}) exitWith {
     playSound ["3DEN_notificationWarning", true];
 };
 
-// Ensure imported items are in scanned config array and classname case is correct
-private _configItems = +(uiNamespace getVariable [QGVAR(configItems), []]);
+params ["_controlsGroup"];
+
+// Convert all items to config case
+_importList = _importList apply {_x call EFUNC(common,getConfigName)};
+
+// Remove any invalid/non-existing items
+_importList = _importList - [""];
+
+// Ensure imported items are in scanned config array
+private _configItems = uiNamespace getVariable [QGVAR(configItems), []];
 private _configItemsFlat = _configItems select [2, 16];
-_configItemsFlat append (_configItems select 0);
-_configItemsFlat append (_configItems select 1);
+_configItemsFlat append (_configItems select IDX_VIRT_WEAPONS);
+_configItemsFlat append (_configItems select IDX_VIRT_ATTACHMENTS);
 
 private _filteredList = [];
+private _intersection = [];
 
 {
-    private _item = _x;
-    {
-        private _index = _x findIf {_x == _item};
-        if (_index > -1) then {
-            _filteredList pushBackUnique (_x select _index);
-        };
-    } forEach _configItemsFlat;
-} forEach _importList;
+    // Find common entries
+    _intersection = _x arrayIntersect _importList;
+
+    _filteredList append _intersection;
+
+    //////////////////////// Is this necessary/wanted?
```

Comment:
Would the code below be of any benefit?

Reply:
It speeds up the array Intersect on next iteration, so yes